src/jenkins_job_insight/diagnostic_archive.py

```
@@ -0,0 +1,476 @@
+        return None
+
+    extract_dir = EXTRACT_BASE / f"diagnostic-archive-{uuid.uuid4().hex[:8]}"
+    extract_dir.mkdir(parents=True, exist_ok=True)
+    logger.info(f"Extracting archive ({size_mb:.1f} MB) to {extract_dir}")
+
+    fileobj = io.BytesIO(archive_data)
+
+    # Try tar first, then zip
+    try:
+        _extract_tar(fileobj, extract_dir)
+    except (tarfile.TarError, Exception):
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)